pkg/provider/branch/cooldown/cooldown.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cooldown
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/k8s"
+	"github.com/go-logr/logr"
+)
+
+// Global variable for CoolDownPeriod allows packages to Get and Set the coolDown period
+var coolDown *cooldown
+
+type cooldown struct {
+	mu sync.RWMutex
```

Comment:
This could also be done with `sync/atomic`. Did not realize earlier that `time.Duration` type is `int64`. 

Perhaps the code would be cleaner with `atomic`, would need to refactor and check this.

Reply:
We can revisit this later as an improvement on all locks.